rewrite-java/src/main/java/org/openrewrite/java/LombokUtilityClass.java

```diff
@@ -0,0 +1,160 @@
+package org.openrewrite.java;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.tree.Flag;
+import org.openrewrite.java.tree.J;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparing;
+
+/**
+ * TODO: Check the following criteria:
+ * - Lombok in dependencies +
+ * - All methods of class are static +
+ * - No instances of given class +
+ * - All static attributes are final +
+ * <p>
+ * TODO: Perform the transformation:
+ * - Add the annotation +
+ * - Remove static from all attributes and methods +
+ * <p>
+ * TODO: Features to consider:
+ * - Transformation: Add Lombok config if not present + supported configuration options for utility class
+ * - Transformation: Replace instantiation with static calls to methods --> na
+ * - Anonymous classes ???
+ * - Reflection ???
+ */
+public class LombokUtilityClass extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Lombok UtilityClass";
+    }
+
+    @Override
+    public String getDescription() {
+        return "This recipe will check if any class is transformable (only static methods in class)" +
+                " into the Lombok UtilityClass and will perform the change if applicable.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new ChangeVisitor();
```

Comment:
You could look at using `Precondition.check` here to first verify that a class is a utility class candidate, before applying the visitor that adds the annotation.